packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -2896,3 +2947,205 @@ typedef SelectableRegionContextMenuBuilder = Widget Function(
   BuildContext context,
   SelectableRegionState selectableRegionState,
 );
+
+/// Signature for the callback that reports when the user changes the selection
+/// under a [SelectionListener].
+///
+/// The [selectionDetails] object provides information about the active
+/// selection contained under the [SelectionListener] this callback is provided to.
+typedef SelectionListenerSelectionChangedCallback = void Function(SelectionDetails selectionDetails);
+
+/// A [SelectionContainer] that allows the user to listen to selection changes
+/// for the child subtree it wraps under a [SelectionArea] or [SelectableRegion].
+///
+/// This widget should have an ancestor [SelectionArea] or [SelectableRegion]
+/// to be able to listen to selection changes in this widgets subtree.
```

Comment:
If it doesn't selection does not work at all, so there are no updates to be received.

Reply:
Would it be possible/desirable to log warning in that case?

Reply:
This isn't possible right now with the current mechanisms. Right now `SelectionRegistrar.maybeOf(context)` is contained under `SelectionContainer`, so that doesn't accurately tell us if there is a `SelectionArea` or not, i.e. a `SelectionContainer` that is wrapped around a `SelectionListener` may not be a `SelectionArea`. I think this is okay for now since `SelectionContainer` itself doesn't really log if it doesn't have an ancestor `SelectionArea`.